src/hip_module.cpp

```diff
@@ -192,7 +192,7 @@ hipError_t ihipModuleLaunchKernel(TlsData *tls, hipFunction_t f, uint32_t global
         if (impCoopParams) {
             const auto p{static_cast<const char*>(*impCoopParams)};
             // The sixth index is for multi-grid synchronization
-            kernargs.insert((kernargs.cend() - padSize - HIP_IMPLICIT_KERNARG_SIZE) + 6 * HIP_IMPLICIT_KERNARG_ALIGNMENT,
+            kernargs.insert((kernargs.cend() - HIP_IMPLICIT_KERNARG_SIZE) + 6 * HIP_IMPLICIT_KERNARG_ALIGNMENT,
```

Comment:
- Using `HIP_IMPLICIT_KERNARG_ALIGNMENT` is odd here.  I think what we really want to say is something like `6 * HIP_IMPLICIT_KERNARG_ELEMENT_SIZE` (even though the alignment and size is the same)?
- `insert` add new elements to the vector and therefore increases its size.  I think you want to `std::copy` into the 6th slot instead?

Reply:
@scchan I removed insert with copy as requested.
Regarding using HIP_IMPLICIT_KERNARG_ELEMENT_SIZE, it is not defined anywhere in HIP.. The byte size and alignment for implicit arguments are the same, so using the  HIP_IMPLICIT_KERNARG_ALIGNMENT should be fine here.